Add and apply cargo bay/shuttle access

warriorstar-orion opened the pull request:
## What Does This PR Do

This PR adds separate access for Cargo Bay and Supply Shuttle, and makes access for several roles a bit more consistent and logical. The main focus is on removing access that remained from when HOP was still in charge of cargo.

In summary, here is everyone's access:

| Job        | Cargo Office | Bay | Supply Shuttle | Mailroom | Mining Dock |
| ---------- | ------------ | --- | -------------- | -------- | ----------- |
| Miner      | ✔️           | ✔️  | ❌             | ✔️       | ✔️          |
| Cargo Tech | ✔️           | ✔️  | ✔️             | ✔️       | ❌          |
| QM         | ✔️           | ✔️  | ✔️             | ✔️       | ✔️          |
| HOP        | ✔️           | ❌  | ❌             | ❌       | ❌          |
| HOS        | ✔️           | ❌  | ❌             | ❌       | ✔️          |

<details><summary>Expand for details</summary>

- adds separate accesses for cargo bay and supply shuttle, and remaps the relevant areas.
- gives miners `ACCESS_CARGO_BAY`. This _only matters on Box_, because miners _already have access to the Cargo Bay on every other map_: it is required to get to the Meta Mining Dock without going through maints, there are no airlocks between Mining Dock and Cargo Bay on Cere, and Delta affords them Cargo Bay access through the north door of Mining Dock. In other words, 3/4 maps already give miners access to Cargo Bay. Box is the odd one out and there's no reason for it to be.
- Removes miner and HOP access to the Supply Shuttle.
- Removes miner and HOP access to the department Supply Shuttle console.
- Adds Cargo Office access to HOS. If they're intended to have mining access, they need to be able to get there. Uncertain if they should have Bay/Shuttle access, because without Bay access they can't get to mining on Meta.
- As a knock-on effect, it now permits miners to access Cargo Office proper on Delta, because again, there's no reason for one map to be the odd one out. Miners have access to Cargo Office on every other map.
</details>

## Why It's Good For The Game

When QM was promoted to Command, HOP's access was supposed to be restricted to "basic", i.e. the Cargo Office proper: https://github.com/ParadiseSS13/Paradise/pull/22133#discussion_r1306731211

Before this PR, the only way to distinguish between "basic" access and access unique to Cargo Techs/QMs was to repurpose the "Mailroom" access, `mail_sorting`. This meant that `mail_sorting` access was used in a whole bunch of places that weren't the mail room, which is confusing, inaccurate, and annoys me on a personal level. In addition, there were places where this _wasn't_ fixed, so the HOP could e.g. access the cargo bay on Box via maintenance. It also resulted in weirdness where e.g. the HOS had access to Mining Dock, but not the Office itself, which is silly.

By separating these accesses out properly, it is much, much easier to reason about who should be able to go where.

Unfortunately, miners have a ton of access compared to techs, because they need access to the Mailroom on Box to get to the ORM, have office access, and have bay access. Not the worst thing but something to think about. Except for Cere, it is, theoretically, possible to remove Cargo Office access for miners, because there's alternate routes on all other maps.

## Testing

- Cerestation
  - [x] QM test
  - [x] HOP test
  - [x] Cargo Tech test
  - [x] Miner test
  - [x] HOS test
- Metastation
  - [x] QM test
  - [x] HOP test
  - [x] Cargo Tech test
  - [x] Miner test
  - [X] HOS test
- Deltastation
  - [x] QM test
  - [x] HOP test
  - [x] Cargo Tech test
  - [x] Miner test
  - [x] HOS test
- Boxstation
  - [x] QM test
  - [x] HOP test
  - [x] Cargo Tech test
  - [x] Miner test
  - [x] HOS test

Note that if you want to test this yourself, spawning `/obj/item/card/id/...` will give you the wrong access. Those have different access lists than the access applied to ID cards when a person spawns with a given job.

<!-- How did you test the PR, if at all? -->

## Changelog

:cl:
tweak: Miners now have Cargo Bay (but not Supply Shuttle) access on all maps.
tweak: Miners now have Cargo Office access on Kerberos.
fix: HOP now properly only has basic access to the Cargo Office.
tweak: HOS now has Cargo Office access on all maps.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->
<!-- If a PR has no impact on players (i.e. a code refactor that does not change functionality) then the entire Changelog heading and contents can be removed. -->

Danchi299 commented:
you should probably also use it for cargo computer, the one that needs access to use

warriorstar-orion commented:
> you should probably also use it for cargo computer, the one that needs access to use

I'm going to circle back to that in a separate PR to keep things simple.

1080pCat commented:
> Job 	Cargo Office 	Bay 	Supply Shuttle 	Mailroom 	Mining Dock
> Miner 	✔️ 	✔️ 	❌ 	✔️ 	✔️
> Cargo Tech 	✔️ 	✔️ 	✔️ 	✔️ 	❌
> QM 	✔️ 	✔️ 	✔️ 	✔️ 	✔️
> HOP 	✔️ 	❌ 	❌ 	❌ 	❌
> HOS 	✔️ 	❌ 	❌ 	❌ 	✔️

bluesheild? how'd they fit in?

Kholdstare2 commented:
Will this affect Paramedic's access to the mining dock?

warriorstar-orion commented:
@1080pCat I adjusted Blueshield access to give them access to what they have now.
@Kholdstare2 I adjusted Paramedic access so they can still reach the mining dock.

Fluffyzack21 commented:
How will this effect NTR access to cargo?

warriorstar-orion commented:
> How will this effect NTR access to cargo?

I just updated NTR access so they can get to the bay and supply shuttle just like prior to this.
